instrumentation/finagle-http-23.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/finaglehttp/v23_11/ChannelTransportInstrumentation.java

```diff
@@ -34,18 +34,18 @@ public void transform(TypeTransformer transformer) {
   public static class WriteAdvice {
 
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void methodEnter(@Advice.Local("otelScope") Scope scope) {
+    public static Object methodEnter() {
```

Comment:
is there a reason for not using return type `Scope` here, and using `@Advice.Enter Scope scope` below? thanks

Reply:
IIRC correctly all the argument and return types need to be accessible from the loaded class:
When inserting the `invokedynamic` bytecode into the instrumented class, it needs to be accompanied with the corresponding [MethodType](https://docs.oracle.com/javase/8/docs/api/java/lang/invoke/MethodType.html) of the method which will be called: This allows the bytecode-verifier to do its job, because the `MethodType` is stored in the constant pool. If that `MethodType` contains unresolvable types, the class won't load.

We can later lift this restriction:
 * When inserting the `invokedynamic`, replace all unaccessible types with `Object`
 * We store the orginial signature before erasure as a `String` in the constant pool and pass it to the `invokedynamic` bootstrap method
 * Using this original signature, the invokedynamic bootstrapping method can lookup the `MethodHandle` for the Advice method and alter it to match the erased signature using the utilities in [MethodHandle.asType](https://docs.oracle.com/javase/8/docs/api/java/lang/invoke/MethodHandle.html#asType-java.lang.invoke.MethodType-).
 
Note that when doing this, we should carefully evaluate whether altering the `MethodHandle` has a negative impact on inlining.

Anyway, I think we should document this limitation / pitfall for now in #11546.

Reply:
The general rule for advice methods should be that their signature should not expose types that are not accessible in the instrumented method. This has been documented in #11546 for modifying method parameters and return values but not for the enter method return value, I'll fix that.

Reply:
I've added this to the documentation : https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11546/commits/547d422eeab1a37f9f88fe33df1bdb731aede34b

Reply:
makes sense, thanks!